### PR TITLE
feat(provider/terminal): add richlink content type support

### PR DESCRIPTION
**Pull request by zelinewang**
## Summary

Adds `richlink` content type support to the `terminal` provider — encode in `spectrumToProtocol`, decode in `protocolToSpectrum`. Closes the `UnsupportedError` gap that surfaced after #29 introduced `richlink` as a first-class content type.

## Background

#29 added `richlink` to spectrum-ts as a content type with lazy async accessors for `title` / `summary` / `cover`. None of the existing providers (iMessage, WhatsApp Business, terminal) handle it yet — they all throw `UnsupportedError`. This PR closes the gap for `terminal`.

## Usage

```typescript
import { Spectrum, richlink } from "spectrum-ts";
import { terminal } from "spectrum-ts/providers/terminal";

const app = await Spectrum({ providers: [terminal.config()] });
for await (const [space] of app.messages) {
  await space.send(richlink("https://example.com/article"));
}
```

## Changes

| File | Change |
|------|--------|
| `packages/spectrum-ts/src/providers/terminal/protocol.ts` | Adds `richlink` variant to `ProtocolContent` (`{ type: "richlink"; url; title?; summary?; cover?: { mimeType?; bytes? } }`). `cover.bytes?` mirrors the `attachment` / `voice` `bytes?` precedent. |
| `packages/spectrum-ts/src/providers/terminal/index.ts` | Adds encode + decode branches mirroring the existing `attachment` / `voice` / `contact` lazy-accessor wrapping pattern. Imports `richlinkSchema` + `RichlinkCover` + `bufferToStream`. Updates the `UnsupportedError` doc-comment example to `effect` (since `richlink` is now supported here). |

## Design notes

**Encode (`spectrumToProtocol`)** — Resolves the lazy accessors eagerly (`Promise.all([title(), summary(), cover()])`), inlines cover bytes as base64. An empty cover buffer (e.g. `buildCover`'s catch-all returns `Buffer.alloc(0)` after a failed image fetch) is dropped to save wire space.

**Decode (`protocolToSpectrum`)** — Wraps the static wire values back into the lazy-accessor shape `richlinkSchema` requires. The accessors are trivial closures — no network call on the receiver, mirroring how `attachment` decode keeps `read()` lazy.

**Wire-boundary normalization** — three runtime safety guards at the decode boundary:

1. `title` / `summary`: normalize `""` → `undefined`. The protocol type permits `""` (JSON has no min-length) but `richlinkSchema` enforces `z.string().min(1).optional()` and Zod v4 validates the accessor *return value* at call time. Without normalization, a peer sending an empty string would defer a `ZodError` into the agent's `await decoded.title()`.
2. `cover.mimeType`: same `""` → `undefined` (same Zod constraint on `richlinkCoverSchema`).
3. `cover.bytes`: optional in the protocol type. Decoder guards `coverWire?.bytes` truthy before constructing the cover; otherwise treats as no-cover (avoids `Buffer.from(undefined, "base64")` crashing on a malformed `cover: { mimeType: "..." }` payload).

## Verification

Local environment: macOS arm64, Bun 1.3.11.

| Tier | Result |
|------|--------|
| `bun run check` (Ultracite / Biome) | ✓ 75 files, 0 issues |
| `bun run build` (tsup ESM + DTS) | ✓ ~3s |
| Contract test (8 scenarios / 19 assertions: full round-trip + empty metadata + empty cover bytes elision + 256-byte binary identity + 4 wire-boundary regression scenarios) | ✓ All pass |
| TCP-level E2E against pinned `tuichat v0.1.4` (real spawn + real LSP-framed JSON-RPC + real `send` with richlink content) | ✓ `SendResult{id, timestamp}`, no `-32099`, exit code 0 |

The wire-boundary normalization fixes were caught by a multi-engine review pass before push (Codex + Claude reviewers + cross-scorer): three independent reviewers flagged the empty-string Zod deferral pattern; one additionally flagged the missing `cover.bytes` guard with a `Buffer.from(undefined)` reproduction. All three fixes were verified by adding regression scenarios to the contract test.

Verification scripts kept locally as artifacts; not part of this PR since the project doesn't have test infrastructure yet — happy to add a `bun:test` setup as a follow-up if welcome.

## Limitations / follow-ups

- **tuichat visual rendering** — `tuichat v0.1.4` (the pinned default binary) accepts the wire shape (verified empirically — no `-32099`) but doesn't yet render `richlink` content visually. `internal/protocol/types.go` `Content` struct has no `Url` / `Title` / `Summary` / `Cover` fields, so unknown wire fields are silently dropped on Go's `Unmarshal`. Filed as photon-hq/tuichat#1 for tracking. The wire-level integration in this PR is forward-compatible — no spectrum-ts-side change needed when tuichat ships rendering.
- **`as SpectrumContent` cast** — My new decode branch ends with `richlinkSchema.parse(...) as SpectrumContent`, consistent with the five existing branches in `protocolToSpectrum`. The project's `.claude/CLAUDE.md` recommends "leverage TypeScript's type narrowing instead of type assertions" — happy to follow up with a separate refactor PR converting all five branches to schema-inferred types, if that direction is welcome.

## Test plan

- [x] `bun run check`
- [x] `bun run build`
- [x] Contract test (round-trip + edge cases + 4 wire-boundary regression)
- [x] TCP-level E2E against real `tuichat v0.1.4`
- [ ] Maintainer review

---
<a href="https://app.blacksmith.sh/photon-hq/codesmith/spectrum-ts/pr/42"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-in-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-in-codesmith-light.svg">View in Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need.</sup>

- [ ] Let Codesmith autofix CI failures and bot reviews

## Summary by CodeRabbit

* **New Features**
  * Richlink content support added to the terminal provider: send and receive rich links with optional title, summary, and cover images. Cover image data is serialized for reliable transmission.

* **Bug Fixes / Resilience**
  * Decoding is more robust: empty or missing fields are normalized and invalid richlink payloads gracefully fall back to the generic payload handling.

### CODE REVIEW

File: packages/spectrum-ts/src/providers/terminal/index.ts [hunk 1]
```diff
@@ -531,6 +560,59 @@ function protocolToSpectrum(p: ProtocolContent): SpectrumContent {
...
+    } else {
+      coverAccessor = () => Promise.resolve(undefined);
+    }
+    return richlinkSchema.parse({
+      type: "richlink",
+      url,
+      title: titleAccessor,
+      summary: summaryAccessor,
+      cover: coverAccessor,
+    }) as SpectrumContent;
```
[the remark this answers is not in the document]
Comment (zelinewang) [reply packages/spectrum-ts/src/providers/terminal/index.ts]: Thanks for catching this — fixed in 8af9400.

Wrapped `richlinkSchema.parse` in try/catch and let the catch fall through to the existing unknown-shape fallback (`{ type: "custom", raw: p }`) at the function tail. This mirrors the contact-decode try/catch at lines 526-532 which already swallows malformed-vCard zod errors for the same reason.

Verified locally with two new contract-test scenarios (now 21 total): malformed `url: "not-a-url"` and missing `url` field both round-trip to a `custom` payload without throwing.

**Comment by underthestars-zhy** [comment]
@zelinewang does this require any changes in tuichat? if doesn't, how can the tuichat render the richlink

**Comment by zelinewang** [comment]
@underthestars-zhy From what I could test against tuichat v0.1.4, the wire round-trips fine. For rendering, yes i opened tuichat#1 for a separate followup on rendering. 

This PR only adds the terminal adapter side, tested with tuichat v0.1.4, the `richlink` send returns `SendResult` instead of `-32099`, but tuichat does not render it yet; For tuichat#1, we need to make `Content` model handle `url` / `title` / `summary` / `cover`

**Comment by underthestars-zhy** [comment]
@zelinewang do u mind also submitting a pr to tuichat so it does support the rendering

**Comment by zelinewang** [comment]
Opened draft PR photon-hq/tuichat#2 for the rendering side. Kept it text-only first and left Pattern A vs Raw passthrough / cover rendering as explicit design questions in the PR body.

**Comment by underthestars-zhy** [comment]
can we also render the image or have a little button that we hover and display and image? like what we have with the current attachmemt

**Comment by zelinewang** [comment]
Yes. I updated photon-hq/tuichat#2 in commit `6c876b4` to model `cover` and reuse the existing attachment preview path: richlinks with `cover.bytes` + a supported image MIME now show `(click to preview)` and open the same floating preview panel as attachments. No spectrum-ts wire change needed.

Verified with `go build ./...`, `go vet ./...`, `go test ./... -race`, plus TCP E2E against spectrum-ts#42: richlink with cover still returns `SendResult` and no `-32099`.
